### PR TITLE
hw 26.1

### PR DESCRIPTION
**Pull request by mikhail771**

### CODE REVIEW

File: src/main/java/com/internet/mixshop/controller/AddProductController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/view/product/add-product.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String price = req.getParameter("price");
```
Comment (dariiabutorina) [inline src/main/java/com/internet/mixshop/controller/AddProductController.java:27]: What if we receive the empty Strings? :)

File: src/main/java/com/internet/mixshop/controller/RegisterController.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        req.getRequestDispatcher("WEB-INF/view/register.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        String pwdConfirm = req.getParameter("psd-repeat");
```
Comment (dariiabutorina) [inline src/main/java/com/internet/mixshop/controller/RegisterController.java:32]: Same for here

**Review by dariiabutorina — CHANGES_REQUESTED** [review]
Well done, let's improve your code just for a bit

File: src/main/java/com/internet/mixshop/controller/AddProductController.java (hunk 1 above)
Comment (RomanDubovskyi) [reply src/main/java/com/internet/mixshop/controller/AddProductController.java:27]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/com/internet/mixshop/controller/AddToCartController.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import com.internet.mixshop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddToCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/controller/AddToCartController.java]: remove empty line

File: src/main/java/com/internet/mixshop/controller/AddToCartController.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import com.internet.mixshop.model.ShoppingCart;
+import com.internet.mixshop.service.ProductService;
+import com.internet.mixshop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddToCartController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/controller/AddToCartController.java]:
```suggestion
public class AddProductToCartController extends HttpServlet {
```

File: src/main/java/com/internet/mixshop/controller/CartController.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import com.internet.mixshop.model.ShoppingCart;
+import com.internet.mixshop.service.ShoppingCartService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CartController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/controller/CartController.java]:
```suggestion
public class GetCartProductsController extends HttpServlet {
```

File: src/main/java/com/internet/mixshop/controller/CartController.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    private static final Long USER_ID = 1L;
+
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/controller/CartController.java]: we don't need here empty line, while declaring fields

File: src/main/java/com/internet/mixshop/controller/DeleteFromCartController.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+import com.internet.mixshop.service.ProductService;
+import com.internet.mixshop.service.ShoppingCartService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteFromCartController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/controller/DeleteFromCartController.java]:
```suggestion
public class DeleteProductFromCartController extends HttpServlet {
```

File: src/main/java/com/internet/mixshop/controller/DeleteUserController.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.mixshop");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String userId = req.getParameter("id");
+        Long id = Long.valueOf(userId);
+        userService.delete(id);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/controller/DeleteUserController.java:24]: let's delete cart here as well

File: src/main/java/com/internet/mixshop/controller/InjectDataController.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        User grinch = new User("Mr.Grinch", "login", "1234");
+        User morty = new User("Morty", "rick", "jessika");
+        userService.create(grinch);
+        userService.create(morty);
+        ShoppingCart cart = new ShoppingCart(grinch.getId());
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/controller/InjectDataController.java:31]: you create cart only for one user

File: src/main/java/com/internet/mixshop/controller/ProductController.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+import com.internet.mixshop.service.ProductService;
+import com.internet.mixshop.service.ShoppingCartService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ProductController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/controller/ProductController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/com/internet/mixshop/controller/RegisterController.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+        if (password.equals(pwdConfirm)) {
+            User user = new User(name, login, password);
+            userService.create(user);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/controller/RegisterController.java:36]: create cart for user here

File: src/main/java/com/internet/mixshop/controller/UserController.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+import com.internet.mixshop.model.User;
+import com.internet.mixshop.service.UserService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class UserController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/controller/UserController.java]:
```suggestion
public class GetAllUsersController extends HttpServlet {
```

File: src/main/webapp/web.xml [hunk 13]
```diff
@@ -6,11 +6,111 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>delete-from-cart</servlet-name>
+        <servlet-class>com.internet.mixshop.controller.DeleteFromCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>delete-from-cart</servlet-name>
+        <url-pattern>/cart/delete</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cart/products/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 14]
```diff
@@ -6,11 +6,111 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>cart</servlet-name>
+        <servlet-class>com.internet.mixshop.controller.CartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>cart</servlet-name>
+        <url-pattern>/cart</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cart/products</url-pattern>
```

File: src/main/webapp/web.xml [hunk 15]
```diff
@@ -6,11 +6,111 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>add-to-cart</servlet-name>
+        <servlet-class>com.internet.mixshop.controller.AddToCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>add-to-cart</servlet-name>
+        <url-pattern>/products/add-to-cart</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cart/products/add</url-pattern>
```

File: src/main/java/com/internet/mixshop/controller/AddProductController.java (hunk 1 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/AddProductController.java:27]: ok

File: src/main/java/com/internet/mixshop/controller/AddToCartController.java (hunk 3 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/AddToCartController.java]: done

File: src/main/java/com/internet/mixshop/controller/AddToCartController.java (hunk 4 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/AddToCartController.java]: done

File: src/main/java/com/internet/mixshop/controller/CartController.java (hunk 5 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/CartController.java]: done

File: src/main/java/com/internet/mixshop/controller/CartController.java (hunk 6 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/CartController.java]: done

File: src/main/java/com/internet/mixshop/controller/DeleteFromCartController.java (hunk 7 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/DeleteFromCartController.java]: done

File: src/main/java/com/internet/mixshop/controller/RegisterController.java (hunk 11 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/RegisterController.java:36]: done

File: src/main/java/com/internet/mixshop/controller/UserController.java (hunk 12 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/UserController.java]: done

File: src/main/webapp/web.xml (hunk 14 above)
Comment (mikhail771) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 15 above)
Comment (mikhail771) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 13 above)
Comment (mikhail771) [reply src/main/webapp/web.xml]: done

File: src/main/java/com/internet/mixshop/controller/DeleteUserController.java (hunk 8 above)
Comment (RomanDubovskyi) [reply src/main/java/com/internet/mixshop/controller/DeleteUserController.java:24]: fix this comments please, rest seem to be good

File: src/main/java/com/internet/mixshop/controller/InjectDataController.java (hunk 9 above)
Comment (RomanDubovskyi) [reply src/main/java/com/internet/mixshop/controller/InjectDataController.java:31]: this comment is not fixed as well

**Review by RomanDubovskyi — COMMENTED** [review]
some of the comments are still not fixed, be more attentive please

File: src/main/java/com/internet/mixshop/controller/DeleteUserController.java (hunk 8 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/DeleteUserController.java:24]: sorry, fixed

File: src/main/java/com/internet/mixshop/controller/InjectDataController.java (hunk 9 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/InjectDataController.java:31]: sorry, fixed

File: src/main/java/com/internet/mixshop/controller/DeleteUserController.java [hunk 16]
```diff
@@ -11,12 +12,15 @@
...
     private UserService userService = (UserService) injector.getInstance(UserService.class);
+    private ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
         String userId = req.getParameter("id");
         Long id = Long.valueOf(userId);
+        shoppingCartService.delete(shoppingCartService.getById(id).getId());
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/controller/DeleteUserController.java]:
```suggestion
        shoppingCartService.delete(shoppingCartService.getByUserId(id).getId());
```

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**